src/lib.rs

```diff
@@ -46,7 +46,8 @@ struct CircomInput {
 }
 
 pub fn create_recursive_circuit(
-    witness_generator_file: PathBuf,
+    witness_generator_file_js: PathBuf,
+    witness_generator_file_wasm: PathBuf,
```

Comment:
actually it would be preferred to keep the C++ witness generator around in some form since WASM is much slower for larger circuits.
Maybe we can make the API take an optional arg `witness_generator_file_wasm` and leave `witness_generator_file` as is now? If `witness_generator_file_wasm` is passed we internally switch between `generate_witness_from_wasm` and `generate_witness_from_bin` logic

Reply:
Makes total sense. I'll update that now and also mention this in the README.